packages/frontend/src/retro/hooks/useTimer.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef, useState } from "react";
+import { useRetroContext } from "../context/RetroContext";
+import { TimerStatus } from "../types/retroTypes";
+
+interface useTimerProps {
+  onTimerFinish: () => void;
+}
+
+export function useTimer({ onTimerFinish }: useTimerProps) {
+  const { timerStatus, timerDuration } = useRetroContext().retroState;
+  const [timeRunning, setTimeRunning] = useState(0);
+  const intervalRef = useRef<NodeJS.Timer>();
+  const remainingTime = timerDuration - timeRunning;
+  const minutes = Math.floor(remainingTime / 1000 / 60);
+  const seconds = Math.round((remainingTime / 1000) % 60);
+  const labelOptions: Intl.DateTimeFormatOptions = {
+    minute: "numeric",
+    second: "numeric",
+    hourCycle: "h23",
+    hour: minutes > 59 ? "numeric" : undefined,
+  };
+  function createLabel() {
+    const date = new Date(0);
+    date.setHours(0, 0, 0);
+    date.setTime(date.getTime() + remainingTime);
+    return new Intl.DateTimeFormat("de-DE", labelOptions).format(date);
```

Comment:
I used a build in option instead of datejs. I think in the current implementation the actions are clearer and we do not need an additional if / else block. The "setHour" seems excessive but is necessary to reach a 23h format